api/v1alpha1/nodemaintenance_types.go

```diff
@@ -20,27 +20,148 @@ import (
+	// to wait before giving up on pod termination, zero means infinite
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=0
+	// +kubebuilder:validation:Minimum:=0
+	TimeoutSecond uint32 `json:"timeoutSeconds,omitempty"`
+}
+
+// DrainSpec describes configuration for node drain during automatic upgrade
+type DrainSpec struct {
+	// Force indicates if force draining is allowed
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	Force bool `json:"force,omitempty"`
 
-	// Foo is an example field of NodeMaintenance. Edit nodemaintenance_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// PodSelector specifies a label selector to filter pods on the node that need to be drained
+	// For more details on label selectors, see:
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+	// +kubebuilder:validation:Optional
+	PodSelector string `json:"podSelector,omitempty"`
+
+	// TimeoutSecond specifies the length of time in seconds to wait before giving up drain, zero means infinite
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=300
+	// +kubebuilder:validation:Minimum:=0
+	TimeoutSecond uint32 `json:"timeoutSeconds,omitempty"`
+
+	// DeleteEmptyDir indicates if should continue even if there are pods using emptyDir
+	// (local data that will be deleted when the node is drained)
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	DeleteEmptyDir bool `json:"deleteEmptyDir,omitempty"`
+
+	// PodEvictionFilters specifies filters for pods that need to undergo eviction during drain.
+	// if specified. only pods that match PodEvictionFilters will be evicted during drain operation.
+	// if unspecified. all non-daemonset pods will be evicted.
+	// logical OR is performed between filter entires. logical AND is performed within different filters
+	// in a filter entry.
+	// +kubebuilder:validation:Optional
+	PodEvictionFilters []PodEvictionFiterEntry `json:"podEvictionFilters,omitempty"`
+}
+
+// PodEvictionFiterEntry defines filters for Pod evictions during drain operation
+type PodEvictionFiterEntry struct {
+	// ByResourceNameRegex filters pods by the name of the resources they consume using regex.
+	ByResourceNameRegex *string `json:"byResourceNameRegex,omitempty"`
 }
 
 // NodeMaintenanceStatus defines the observed state of NodeMaintenance
 type NodeMaintenanceStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-	Bar string `json:"bar,omitempty"`
+	// Conditions represents observations of NodeMaintenance current state
+	// +kubebuilder:validation:Optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
```

Comment:
the linter is not happy with formation for this string

Reply:
fixed